rholang/src/main/scala/coop/rchain/rholang/interpreter/matcher/StreamT.scala

```diff
@@ -189,6 +190,29 @@ trait StreamTInstances2 {
 
       def zero[A](state: Stream[A]): Boolean = state.isEmpty
     }
+
+  implicit def streamTSync[F[_]](implicit F0: Sync[F]): Sync[StreamT[F, ?]] =
+    new StreamTSync[F]() {
+      implicit val F = F0
+    }
+}
+
+private trait StreamTSync[F[_]] extends Sync[StreamT[F, ?]] with StreamTMonadError[F, Throwable] {
```

Comment:
Please remember to add laws tests for this Sync instance. See StreamTLawsSpec. The laws are defined on https://github.com/typelevel/cats-effect/blob/master/laws/shared/src/main/scala/cats/effect/laws/discipline/SyncTests.scala. We seem to need a cats-effect-laws dependency

Reply:
I'm afraid it is impossible to construct a lawful `Sync` instance for `StreamT`. It will always be a bit special - that is - possible to use only when the stream is not empty. Otherwise it won't be possible to `acquire` an element from it and provide it to the `use` function

Reply:
I was wrong

Reply:
Then I guess it's not a good idea to construct a Sync[StreamT] instance